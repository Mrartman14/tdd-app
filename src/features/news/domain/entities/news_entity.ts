interface INewsEntity {
    title: string,
    text: string,
}

interface INewsEntityProps {
    title: string,
    text: string,
}

class NewsEntity implements INewsEntity {
    title: string;
    text: string;

    constructor(props: INewsEntityProps) {
        this.title = props.title;
        this.text = props.text;
    }
}

export {
    NewsEntity,
};