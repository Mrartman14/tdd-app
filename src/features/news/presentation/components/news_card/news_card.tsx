import React from 'react';

import { NewsEntity } from '../../../domain/entities/news_entity';

import './news_card.css';

interface NewCardProps {
    data: NewsEntity;
}

class NewsCard extends React.PureComponent<NewCardProps> {
    render() {
        return (
            <div className="news-card">
                <div className="news-card__title">
                    {this.props.data.title}
                </div>
                <div className="news-card__text">
                    {this.props.data.text}
                </div>
            </div>
        );
    }
}

export {
    NewsCard,
};