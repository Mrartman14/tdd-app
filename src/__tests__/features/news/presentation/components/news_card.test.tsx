import { shallow, configure, ShallowWrapper } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { act } from 'react-dom/test-utils';

import { NewsEntity } from '../../../../../features/news/domain/entities/news_entity';
import { NewsCard } from '../../../../../features/news/presentation/components/news_card/news_card';

configure({
    adapter: new Adapter(),
});

const fakeData = new NewsEntity({
    title: 'test title',
    text: 'test text',
});

function setupComponent(props: NewsEntity) {
    const component = shallow(<NewsCard data={props} />);
    return component;
}

describe('should newscard renders well', () => {
    let component: ShallowWrapper;

    beforeEach(() => {
        component = setupComponent(fakeData);
    });
    
    afterEach(() => {
        //
    });

    it('should render a root component element', () => {
        let rootEl: ShallowWrapper;
        act(() => {
            rootEl = component.find('.news-card');
        });
        expect(rootEl!.length).toBe(1);
    });

    it('should render a title', () => {
        let title: ShallowWrapper;
        act(() => {
            title = component.find('.news-card__title');
        });
        expect(title!.text()).toContain(fakeData.title);
    });

    it('should render a text', () => {
        let textContainer: ShallowWrapper;
        act(() => {
            textContainer = component.find('.news-card__text');
        });
        expect(textContainer!.text()).toBe(fakeData.text);
    });

    it('snapshot', () => {
        expect(component).toMatchSnapshot();
    });
});

