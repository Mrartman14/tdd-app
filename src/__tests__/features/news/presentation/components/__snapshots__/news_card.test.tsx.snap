// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should newscard renders well snapshot 1`] = `
<div
  className="news-card"
>
  <div
    className="news-card__title"
  >
    test title
  </div>
  <div
    className="news-card__text"
  >
    test text
  </div>
</div>
`;
